mathoxide-lib/src/views.rs

```diff
@@ -4,6 +4,18 @@ pub trait ArrayView {
     type IterType;
 
     fn translate<ListType: AsRef<[usize]>>(&self, idx: ListType) -> usize;
+    fn try_translate<ListType: AsRef<[usize]>>(&self, idx: ListType) -> Result<usize, &str> {
```

Comment:
Maybe `checked_translate` might be a better name. I think `try_` prefixes are added in functions that panic, when you want a function that returns a `Result` instead. On the other hand there are `checked_` prefixed methods in the standard library that perform some sanity checks before computing a value like this one [here](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_div). However this particular one returns `Option` instead of result, so we would be breaking the convention anyway.

So I  guess up to you what you wanna keep, just presenting an alternative that seems better to me.